client/field_kind.go

```diff
@@ -102,7 +84,7 @@ type SelfKind struct {
 
 var _ FieldKind = ScalarKind(0)
 var _ FieldKind = ScalarArrayKind(0)
-var _ FieldKind = (*SchemaKind)(nil)
+var _ FieldKind = (*CollectionKind)(nil)
 var _ FieldKind = (*SelfKind)(nil)
```

Comment:
todo: The documentation on `SelfKind.RelativeID` is not good enough, a reviewer had to ask me for help on discord.

- [ ] Expand `SelfKind.RelativeID` docs

Reply:
Leaving as-is unless pushback received, the `SelfKind` documentation covers this.